taca/delivery/delivery_classes.py

```diff
@@ -389,3 +393,70 @@ class UploadElement(Upload):
 
     def __init__(self, project, stage_dir):
         super().__init__(project, stage_dir)
+
+
+class Release:
+    """Defines a generic release object."""
+
+    def __init__(self, project, dds_project, dds_deadline, no_dds_mail):
+        self.project_id = project
+        self.dds_project = dds_project
+        self.dds_deadline = dds_deadline
+        self.no_dds_mail = no_dds_mail
+
+    def release_project(self):
+        """Release DDS project to user."""
+        try:
+            cmd = [
+                "dds",
+                "--no-prompt",
+                "project",
+                "status",
+                "release",
+                "--project",
+                self.dds_project,
+                "--deadline",
+                str(self.dds_deadline),
+            ]
+            if self.no_dds_mail:
+                cmd.append("--no-mail")
+            process_handle = subprocess.run(cmd)
+            process_handle.check_returncode()
+            logger.info(
+                f"Project {self.project_id} succefully delivered. Delivery project {self.dds_project} was released to the user."
+            )
+        except subprocess.CalledProcessError as e:
+            logger.error(
+                f"Could not release project {self.project_id}, an error occurred."
+            )
+            raise e
+
+
+class ReleaseNanopore(Release):
```

Comment:
What would be the difference between the different Release classes?

Reply:
Updating statusdb with the delivery status (in the future). For ONT the status would be updated on the project/FC level but for Element and Illumina it would be updated also on the sample level. Probably.

Reply:
👍 I can see that